client/sources/common/interpreter.py

```diff
@@ -253,7 +255,15 @@ def _compare(self, expected, code):
                 actual = printed.strip()
 
         expected = expected.strip()
-        if expected != actual:
+        
+        if not self.skip_locked_cases and expected != actual:
+            actual = hmac.new(self.hash_key.encode('utf-8'),
```

Comment:
Now that this locking procedure is being used in multiple places, you should move it into its own file into a common directory (probably `client/utils`, since this is used in both `client/sources` and `client/protocols`):

1. Better abstraction: callers of the util won't (and shouldn't) need to know that locking is a hash using `hmac`.
2. Easier to maintain: if we ever change the locking implementation, we would only need to change it in `utils` instead of across several files.

Off the top of my head, I think this locking is being used in LockProtocol, UnlockProtocol, and here. You can also use the new locking util in the tests you wrote.